wazo_dird/database/queries/phonebook.py

```diff
@@ -40,7 +40,6 @@ class PhonebookDict(TypedDict):
 
 
 def phonebook_key_to_filter(phonebook_key: PhonebookKey) -> ColumnElement:
-    assert 'id' in phonebook_key or 'uuid' in phonebook_key
```

Comment:
This was discussed with @fblackburn1 in previous PR review.

I like using assert to signal expectations about data that are not explicit(e.g. when type annotations cannot account for them) and that should hold for the code to make sense. It's more a comment to communicate expectations to other programmers. And type checkers can use assert statements for type narrowing(https://mypy.readthedocs.io/en/stable/type_narrowing.html).
These assertions are not used to catch expected error conditions, but to provide sanity checks that clarify understanding of the system.

 There are no donwsides I can see to having asserts in production code, as long as they do not replace what should be expected error conditions(specific exception that should be raised and potentially handled).
 
If the assert is triggered in production, it signals the code is broken and needs to be redesigned around that situation. If the assert would have been triggered but has been optimized out in production, the code will usually fail in another obvious way(such as here, when accessing the 'id' or 'uuid' key). The assert is a shortcut to make clear and explicit the expectations of the code and intent of the programmer. Short and explicit is better than long/complex and implicit.

 Removing assertions without replacing them with a proper alternative simply removes information.